x-pack/plugins/ml/public/file_datavisualizer/components/edit_flyout/overrides_validation.js

```diff
@@ -0,0 +1,131 @@
+  const result = { isValid: true, errorMessage: null };
+
+  if (timestampFormat.indexOf('\n') >= 0 || timestampFormat.indexOf('\r') >= 0) {
+    result.isValid = false;
+    result.errorMessage = i18n.translate('xpack.ml.fileDatavisualizer.editFlyout.overrides.customTimestampValidationErrorMessage', {
+      defaultMessage: 'Multi-line timestamp formats {timestampFormat} not supported',
+      values: {
+        timestampFormat,
+      }
+    });
+    return result;
+  }
+
+  if (timestampFormat.indexOf(INDETERMINATE_FIELD_PLACEHOLDER) >= 0) {
+    result.isValid = false;
+    result.errorMessage = i18n.translate('xpack.ml.fileDatavisualizer.editFlyout.overrides.customTimestampValidationErrorMessage', {
+      defaultMessage: 'Timestamp format {timestampFormat} not supported because it contains {fieldPlaceholder}',
+      values: {
+        timestampFormat,
+        fieldPlaceholder: INDETERMINATE_FIELD_PLACEHOLDER,
+      }
+    });
+    return result;
+  }
+
+  let notQuoted = true;
+  let prevChar = null;
+  let prevLetterGroup = null;
+  let pos = 0;
+
+  while (pos < timestampFormat.length) {
+    const curChar = timestampFormat.charAt(pos);
+
+    if (curChar === '\'') {
+      notQuoted = !notQuoted;
+    } else if (notQuoted && isLetter(curChar)) {
+      const startPos = pos;
+      let endPos = startPos + 1;
+      while (endPos < timestampFormat.length && timestampFormat.charAt(endPos) === curChar) {
+        ++endPos;
+        ++pos;
+      }
+
+      const letterGroup = timestampFormat.substring(startPos, endPos);
+
+      if (VALID_LETTER_GROUPS[letterGroup] !== true) {
+        // Special case of fractional seconds
+        if (curChar !== 'S' || FRACTIONAL_SECOND_SEPARATORS.indexOf(prevChar) === -1 ||
+          !('ss' === prevLetterGroup) || endPos - startPos > 9) {
+          let msg = 'Letter group {letterGroup} in {timestampFormat} is not supported';
+          if (curChar === 'S') {
+            msg += ' because it is not preceeded by [ss] and a separator from {separators}';
+          }
+
+          const values = {
+            timestampFormat,
+            letterGroup
+          };
+
+          if (msg.indexOf('separator') !== -1) {
+            values.separators = FRACTIONAL_SECOND_SEPARATORS;
+          }
+
+          result.isValid = false;
+          result.errorMessage = i18n.translate('xpack.ml.fileDatavisualizer.editFlyout.overrides.customTimestampValidationErrorMessage', {
+            defaultMessage: msg,
+            values,
+          });
+
+          return result;
+        }
+      }
+      prevLetterGroup = letterGroup;
+    }
+
+    prevChar = curChar;
+    ++pos;
+  }
+
+  if (prevLetterGroup == null) {
+    result.isValid = false;
+    result.errorMessage = i18n.translate('xpack.ml.fileDatavisualizer.editFlyout.overrides.customTimestampValidationErrorMessage', {
+      defaultMessage: 'No time format letter groups in override format {timestampFormat}',
```

Comment:
Replace "override" with "timestamp" in this message to be consistent with the other messages.

Reply:
Done in 2adb1002d6ef80fb467803db00d1273545636951